dotnet/src/SemanticKernel.Abstractions/Events/FunctionInvokingEventArgs.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.ComponentModel;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Events;
+
+/// <summary>
+/// Event arguments available to the Kernel.FunctionInvoking event.
+/// </summary>
+public sealed class FunctionInvokingEventArgs : CancelEventArgs
+{
+    internal FunctionInvokingEventArgs(FunctionView functionView, SKContext context, string? prompt)
+    {
+        Verify.NotNull(context);
+
+        this.FunctionView = functionView;
+        this.SKContext = context;
+        this.Prompt = prompt;
```

Comment:
Might be worth including on the FunctionInvokedEventArgs as well? That way folks don't have to register both if they don't intend on modifying a prompt or context before execution.

Reply:
Done, good point!